includes/specials/SpecialUpload.php

```diff
@@ -110,7 +110,7 @@ protected function loadRequest() {
 
 		// If it was posted check for the token (no remote POST'ing with user credentials)
 		$token = $request->getVal( 'wpEditToken' );
-		$this->mTokenOk = $this->getUser()->matchEditToken( $token );
+		$this->mTokenOk = !is_null( $token ) && $this->getUser()->matchEditToken( $token ); # Wikia change
```

Comment:
I'd add this check just before call to `hash_equals()` and return `false`.
We'd get more cases covered instead this one.
Any opinion?